src/Detector/Node/NodeDetector.cs

```diff
@@ -211,15 +213,20 @@ private IEnumerable<FrameworkInfo> DetectFrameworkInfos(DetectorContext context)
 
                 // wild-card dependency
                 (bool isWildCardDependency, string wildCarddependencyName) = GetWildCardDependency(dependencyName);
-
-                if (monitoredDevDependencies.ContainsKey(dependencyName) || isWildCardDependency) 
+                
+                bool isValidDependency = monitoredDevDependencies.ContainsKey(dependencyName);
```

Comment:
nit: we should be able to remove the need for this `isValidDependency` variable by checking for the right-hand side in a conditional and continuing through the loop if it doesn't hold true (since we don't take any actions if `isValidDependency` is `false`):

```cs
// Ignore invalid dev dependencies
if (!monitoredDevDependencies.ContainsKey(dependencyName))
{
    continue;
}
```

Reply:
I've added a work-item 1493329 to refactor later. I've added points to consider in the story description